src/sign.c

```diff
@@ -22,88 +22,87 @@ typedef struct sign sign_T;
 
 struct sign
 {
-    sign_T	*sn_next;	// next sign in list
-    int		sn_typenr;	// type number of sign
-    char_u	*sn_name;	// name of sign
-    char_u	*sn_icon;	// name of pixmap
-# ifdef FEAT_SIGN_ICONS
-    void	*sn_image;	// icon image
-# endif
-    char_u	*sn_text;	// text used instead of pixmap
-    int		sn_line_hl;	// highlight ID for line
-    int		sn_text_hl;	// highlight ID for text
-    int		sn_cul_hl;	// highlight ID for text on current line when 'cursorline' is set
-    int		sn_num_hl;	// highlight ID for line number
-    int		sn_priority;	// default priority of this sign, -1 means SIGN_DEF_PRIO
+    sign_T *sn_next; // next sign in list
+    int sn_typenr;   // type number of sign
+    char_u *sn_name; // name of sign
+    char_u *sn_icon; // name of pixmap
+#ifdef FEAT_SIGN_ICONS
+    void *sn_image;  // icon image
+#endif
+    char_u *sn_text; // text used instead of pixmap
+    int sn_line_hl;  // highlight ID for line
+    int sn_text_hl;  // highlight ID for text
+    int sn_cul_hl; // highlight ID for text on current line when 'cursorline' is set
+    int sn_num_hl;   // highlight ID for line number
+    int sn_priority; // default priority of this sign, -1 means SIGN_DEF_PRIO
 };
 
-static sign_T	*first_sign = NULL;
-static int	next_sign_typenr = 1;
+static sign_T *first_sign = NULL;
+static int next_sign_typenr = 1;
 
 static void sign_list_defined(sign_T *sp);
 static void sign_undefine(sign_T *sp, sign_T *sp_prev);
 
-static char *cmds[] = {
-			"define",
-# define SIGNCMD_DEFINE	0
-			"undefine",
-# define SIGNCMD_UNDEFINE 1
-			"list",
-# define SIGNCMD_LIST	2
-			"place",
-# define SIGNCMD_PLACE	3
-			"unplace",
-# define SIGNCMD_UNPLACE 4
-			"jump",
-# define SIGNCMD_JUMP	5
-			NULL
-# define SIGNCMD_LAST	6
+static char *cmds[] = { "define",
+#define SIGNCMD_DEFINE 0
+                        "undefine",
+#define SIGNCMD_UNDEFINE 1
+                        "list",
+#define SIGNCMD_LIST 2
+                        "place",
+#define SIGNCMD_PLACE 3
+                        "unplace",
+#define SIGNCMD_UNPLACE 4
+                        "jump",
+#define SIGNCMD_JUMP 5
+                        NULL
+#define SIGNCMD_LAST 6
 };
 
-#define FOR_ALL_SIGNS(sp)	\
+#define FOR_ALL_SIGNS(sp) \
     for ((sp) = first_sign; (sp) != NULL; (sp) = (sp)->sn_next)
 
-static hashtab_T	sg_table;	// sign group (signgroup_T) hashtable
-static int		next_sign_id = 1; // next sign id in the global group
+static hashtab_T sg_table;   // sign group (signgroup_T) hashtable
+static int next_sign_id = 1; // next sign id in the global group
 
-/*
+                             /*
  * Initialize data needed for managing signs
  */
-    void
+void
 init_signs(void)
 {
-    hash_init(&sg_table);		// sign group hash table
+    hash_init(&sg_table); // sign group hash table
 }
 
 /*
  * A new sign in group 'groupname' is added. If the group is not present,
  * create it. Otherwise reference the group.
  */
-    static signgroup_T *
+static signgroup_T *
```

Comment:
The Vim coding convention is to use one 'shiftwidth' for the function return type.

Reply:
At the end of the description of the issue I acknowledge this and the main problem is that no formatter support this kind of formatting (at least the one I've tried)

> * Return types will no longer be indented by one 'shiftwidth'
>   * I couldn't find and option to do so but I think having a formatter is better that having indented return types.

In my opinion this is not a blocker for having a formatter because will improve signicantly the consistence of the code